framework-plugins/lisk-framework-chain-connector-plugin/src/db.ts

```diff
@@ -156,4 +162,64 @@ export class ChainConnectorStore {
 		const encodedInfo = codec.encode(ccmsFromEventsSchema, { ccmsFromEvents: ccms });
 		await this._db.set(DB_KEY_CROSS_CHAIN_MESSAGES, encodedInfo);
 	}
+
+	public async getLastSentCCM(): Promise<LastSentCCMWithHeight | undefined> {
+		let lastSentCCM: LastSentCCMWithHeight | undefined;
+		try {
+			const encodedInfo = await this._db.get(DB_KEY_LAST_SENT_CCM);
+			lastSentCCM = codec.decode<LastSentCCMWithHeight>(lastSentCCMWithHeight, encodedInfo);
+		} catch (error) {
+			if (!(error instanceof liskDB.NotFoundError)) {
+				throw error;
+			}
+		}
+		return lastSentCCM;
+	}
+
+	public async setLastSentCCM(lastSentCCM: LastSentCCMWithHeight) {
+		await this._db.set(DB_KEY_LAST_SENT_CCM, codec.encode(lastSentCCMWithHeight, lastSentCCM));
+	}
+
+	public async getCertificates(): Promise<Certificate[]> {
+		let certificates: Certificate[] | undefined = [];
```

Comment:
In which situation would `certificates` be undefined? If `.get` has error it will be caught and return `[]` instead?

Reply:
thats a mistake